internal/olm/operator/registry/index_image.go

```diff
@@ -124,14 +151,319 @@ func (c IndexImageCatalogCreator) CreateCatalog(ctx context.Context, name string
+	bundleDeclConfig, err := render.Run(ctx)
+	log.SetOutput(os.Stdout)
+
+	if len(bundleDeclConfig.Bundles) != 1 {
+		return "", errors.New("bundle image must have at least one bundle")
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	// search for the specific channel in which the upgrade needs to take place, and upgrade from the channel head
+	for i, _ := range originalDeclCfg.Channels {
+		if originalDeclCfg.Channels[i].Name == channelName && originalDeclCfg.Channels[i].Package == bundleDeclConfig.Bundles[0].Package {
+			// found specific channel
+			channelHead, err := getChannelHead(originalDeclCfg.Channels[i].Entries)
+			if err != nil {
+				return "", err
+			}
+			entry := declarativeconfig.ChannelEntry{
+				Name:     bundleDeclConfig.Bundles[0].Name,
+				Replaces: channelHead,
+			}
+			originalDeclCfg.Channels[i].Entries = append(originalDeclCfg.Channels[i].Entries, entry)
+			break
+		}
+	}
+
+	// add the upgraded bundle to resulting declarative config
+	originalDeclCfg.Bundles = append(originalDeclCfg.Bundles, bundleDeclConfig.Bundles[0])
+
+	// convert declarative config to string
+	content, err := StringifyDecConfig(originalDeclCfg)
+
+	if err != nil {
+		log.Errorf("error converting declarative config to string: %v", err)
+		return "", err
+	}
+
+	if content == "" {
+		return "", errors.New("File-Based Catalog contents cannot be empty")
+	}
+
+	if err = ValidateFBC(originalDeclCfg); err != nil {
+		log.Errorf("error validating the generated FBC: %v", err)
+		return "", err
+	}
+
+	log.Infof("Generated a valid Upgraded File-Based Catalog")
+
+	return content, nil
+}
+
+// setupFBCupdates starts the process of upgrading a bundle in an FBC. This function will recreate the FBC that was generated
+// during run bundle and upgrade a specific bundle in the specified channel.
+func setupFBCupdates(c *IndexImageCatalogCreator, ctx context.Context) error {
+	var (
+		originalDeclcfg *declarativeconfig.DeclarativeConfig
+		err             error
+		render          action.Render
+	)
+
+	// render the index image if it is not the default index image
+	render = action.Render{Refs: []string{}}
+	if c.IndexImage != DefaultIndexImage {
+		render.Refs = append(render.Refs, c.IndexImage)
+	}
+
+	log.SetOutput(ioutil.Discard)
+	originalDeclcfg, err = render.Run(ctx)
+	log.SetOutput(os.Stdout)
+
+	if err != nil {
+		return err
+	}
+
+	// add the upgraded bundle to the list of previous bundles so that they can be rendered with a single API call
+	c.PreviousBundles = append(c.PreviousBundles, c.BundleImage)
+	f := &FBCContext{
+		Package:     c.PackageName,
+		Refs:        c.PreviousBundles,
+		ChannelName: c.ChannelName,
+	}
+
+	// Adding the FBC "f" to the originalDeclcfg to generate a new FBC
+	declcfg, err := upgradeFBC(f, originalDeclcfg, ctx)
+	if err != nil {
+		log.Errorf("error creating the upgraded FBC: %v", err)
+		return err
+	}
+
+	// convert declarative config to string
+	content, err := StringifyDecConfig(declcfg)
```

Comment:
If it's not too late, I think this should be `StringifyDeclConfig` because use `declcfg` or `declconfig` in other places. `DecConfig` looks like decimal or december :)

Reply:
That's fair, I can change that.